server/src/plugins/tx-monitor.ts

```diff
@@ -0,0 +1,69 @@
+import fp from 'fastify-plugin';
+import { FastifyInstance } from 'fastify';
+import { IncomingMessage, ServerResponse, Server } from 'http';
+
+declare module 'fastify' {
+  export interface FastifyInstance<
+    HttpServer = Server,
+    HttpRequest = IncomingMessage,
+    HttpResponse = ServerResponse
+  > {
+    authenticate: any;
+  }
+}
+
+const cron = require('node-cron');
+const fetch = require('node-fetch');
+
+import { getPendingTxs, updateTransactionStatus } from '../db';
+import { TransactionStatus } from '../types';
+
+const EXPLORER_API_BASE = 'https://api.blockcypher.com/v1';
+
+export default fp(function transactionsMonitorCron(
+  fastify: FastifyInstance<Server, IncomingMessage, ServerResponse>,
+  opts,
+  next
+) {
+    const monitor = async () => {
+        const pendingTransactions = await getPendingTxs();
+        if (!pendingTransactions || pendingTransactions.length === 0) return;
+        ​
+        const txPromises = pendingTransactions.map(async ({ tx_hash: txHash }) => {
+            const json: any = await getTxStatus(txHash);
```

Comment:
Great! Optionally we can improve the typings here since you are using a generic `<A>` on your function definition over line 61:

```javascript
async function getTxStatus<A>(hash: string): Promise<A> {
```

1. First let's define a type `TransactionStatus`. You can place it were you find it most useful I would recommend over `src/types.ts` since all types are defined there.
```typescript
type TransactionStatus = {
  confirmed: boolean,
  execution_error: boolean
}
```

2. Next let's use this type:
```typescript
const json = await getTxStatus<TransactionStatus>(txHash);
```

<img width="714" alt="Captura de Pantalla 2019-10-02 a la(s) 10 02 51 a  m" src="https://user-images.githubusercontent.com/2475912/66046284-d4d27f80-e4fb-11e9-9f9e-09ddf96bafeb.png">